packages/flutter_tools/lib/src/commands/format.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:path/path.dart' as path;
+
+import '../base/process.dart';
+import '../cache.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class FormatCommand extends FlutterCommand {
+  @override
+  final String name = 'format';
+
+  @override
+  final String description = 'Format one or more dart files.';
+
+  @override
+  bool get requiresProjectRoot => false;
+
+  @override
+  String get invocation => "${runner.executableName} $name <one or more paths>";
+
+  @override
+  Future<int> runInProject() async {
+    if (argResults.rest.isEmpty) {
+      printStatus('No files specified to be formatted.');
+      printStatus(usage);
+      return 1;
+    }
+
+    String dartfmt = path.join(
+        Cache.flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', 'dartfmt');
+    List<String> cmd = <String>[dartfmt, '-w']..addAll(argResults.rest);
```

Comment:
flutter_tools references the analyzer package, and dart_style references the analyzer package. If we take a dependency on dart_style, then when we want to update to a new analyzer version, we may need to first wait until dart_style is updated to support that new version.

There's a brief discussion of the shared dependency issue here: https://www.dartlang.org/tools/pub/versioning#version-lock.